lib/tty/reader.rb

```diff
@@ -256,7 +256,7 @@ def read_line(prompt = "", **options)
         elsif console.keys[char] == :up
           line.replace(history_previous) if history_previous?
         elsif console.keys[char] == :down
-          line.replace(history_next? ? history_next : "")
+          line.replace(history_next? ? history_next : "") if track_history?
```

Comment:
How about replicating the `:up` behaviour?

```ruby
line.replace(history_next) if history_next?
```
Do you see any downsides?

Reply:
I think you want that behavior when the history is enabled – for instance, in zsh, scrolling down "past the end" of the history does indeed give you a fresh blank command-line, which I think is a good idea.

Reply:
Ah, great point!